meeteval/viz/visualize.js

```diff
@@ -1624,10 +1642,37 @@ class CanvasPlot {
         formatValue(element, key, value) {
             if (/^([a-zA-Z0-9_/.-]+\.(wav|flac)(::\[[\d.:]+])?)$/.test(value)) {
                 // Audio path: Display audio player
+
+                let trials = []
+                let url = new URL(value, window.location.href);
+                if (window.location.protocol == 'file:') {
+                    if (value[0] === '/'){
+                        trials.push(settings.audio_server + value);
+                        trials.push("file:////" + value);
+                    } else {
+                        trials.push(settings.audio_server + url.pathname);
+                        trials.push(settings.audio_server + value);  // usually fails, but audio_server could define a prefix
+                        trials.push("file:////" + url.pathname);
+                        trials.push("file:////" + value);  // Can this work?
+                    }
+                } else {  // http or https
+                    // file:// is not allowed in the browser for http(s), so we can't use it
+                    if (value[0] === '/'){
+                        trials.push(settings.audio_server + value);
+                        trials.push(url.href);
+                    } else {
+                        trials.push(settings.audio_server + value);  // usually fails, but audio_server could define a prefix
+                        // trials.push(settings.audio_server + url.pathname);  // could this work?
+                        trials.push(url.href);
+                    }
+                }
+
+                console.log("Audio candidates", trials)
+
```

Comment:
This code should be moved to a function and also used in `drawRecordingAudioButton`

Reply:
drawRecordingAudioButton is different, it requires the audio_server. I don't know if and how to partially load audio.

Additionally, drawRecordingAudioButton is more verbose, so it feels more natural to try only one combination.

Reply:
Hmm, okay. As a user, I would expect that both work the same. But you are right, as far as we know, partial loading is not possible without a server